src/avro_util.erl

```diff
@@ -560,7 +560,8 @@ is_valid_name_head(S) ->
 
 %% @private In addition to what applies to leading char, body char can be 0-9.
 is_valid_name_char(S) -> is_valid_name_head(S) orelse
-                         (S >= $0 andalso S =< $9).
```

Comment:
Should this additional check be added to `is_valid_name_head` instead? Looks like that's where `_` is being allowed too?

https://github.com/carsdotcom/erlavro/blob/afb9accbd86a2a7c7aee47fc6b5066e09b097668/src/avro_util.erl#L559

Reply:
I made an assumption that atoms cannot start with hypens and I was wrong. It also appears that kafka topics can start with hyphens, too.  https://github.com/apache/kafka/blob/0.10.2/core/src/main/scala/kafka/common/Topic.scala#L24

Reply:
Moved it to `is_valid_name_head/1`

Reply:
Now that you mention it, it's wild that a leading dash is still a valid topic name. The Scala code doesn't lie though.